command/commands.go

```diff
@@ -905,9 +908,17 @@ func initCommands(ui, serverCmdUi cli.Ui, runOpts *RunOptions) map[string]cli.Co
 	}
 
 	entInitCommands(ui, serverCmdUi, runOpts, commands)
+	initHCPCommands(ui, commands)
+
 	return commands
 }
 
+func initHCPCommands(ui cli.Ui, commands map[string]cli.CommandFactory) {
+	for cmd, cmdFactory := range hcpengine.InitHCPCommand(ui) {
+		commands[cmd] = cmdFactory
```

Comment:
Should we check for conflicts here?

Reply:
What do you mean? Using the ok idiom to check if the map already has the key?

Reply:
But why would we do that? I think we would like the value to be overridden in case it exists.

Reply:
Since this isn't something actionable for users, I don't think we should show any errors when there's a conflict. What do you think?

Reply:
```
	for cmd, cmdFactory := range hcpvlib.InitHCPCommand(ui) {
		// check for conflicts and only put command in the map in case it doesn't conflict with existing one
		_, ok := commands[cmd]
		if !ok {
			commands[cmd] = cmdFactory
		}
	}
```

Reply:
I feel that we should either log the conflict or return an error.

Reply:
If it returned an error, we could unit test it.

Reply:
in order to return an error or log an error, I would have to change the entire initCommands function structure and signature. I printed the error instead. What do you think?

Reply:
There isn't much the user can do, but I can test it with a mocked ui

Reply:
initCommands is only called in one place, so adding an error to its return shouldn't be a big deal.  You don't have to modify everything else, just to have it handle an error returned from initHCPCommands.

Reply:
> initCommands is only called in one place, so adding an error to its return shouldn't be a big deal. You don't have to modify everything else, just to have it handle an error returned from initHCPCommands.

Handling the error in the RunCustom function also requires initializing a new logger which would write to stderr. I can do that if you think my proposed solution isn't the ideal.

Reply:
Hmm, maybe instead of thinking of this in terms of errors exposed to users, we should have something commands_test that breaks the build if a conflict is introduced.  e.g. add error response value to initCommands; have initHCPCommands populate that error on conflict; have the test call initCommands and look for a non-nil error.    Then RunCustom can ignore the error (with a comment explaining that the error is only used in that test.)

Reply:
Hmmm I see what you'd like to have. I can change that. :)